File: app/src/main/java/com/iita/akilimo/utils/enums/EnumCassavaProduceType.java
package com.iita.akilimo.utils.enums;

public enum EnumCassavaProduceType {
    @Deprecated
    GARI {
        @Override
        public String produce() {
            return "gari";
        }
    },
    @Deprecated
    FLOUR {
        @Override
        public String produce() {
            return "flour";
        }
    },
    ROOTS {
        @Override
        public String produce() {
            return "roots";
        }
    },
    @Deprecated
    CHIPS {
        @Override
        public String produce() {
            return "chips";
        }
    };

    public abstract String produce();
}

File: app/src/main/java/com/iita/akilimo/utils/enums/EnumCassavaProduceType.kt
package com.iita.akilimo.utils.enums

enum class EnumCassavaProduceType {
    GARI {
        override fun produce(): String {
            return "gari"
        }
    },
    FLOUR {
        override fun produce(): String {
            return "flour"
        }
    },
    ROOTS {
        override fun produce(): String {
            return "roots"
        }
    },
    CHIPS {
        override fun produce(): String {
            return "chips"
        }
    };

    abstract fun produce(): String
}
